### ISSUE TITLE
Service Provider not find resources files, because is not loaded correctly

### ISSUE THREAD

**Issue by BoGnY** [rinvex/laravel-subscriptions#86]
This is the console log:

```shell script
:$ composer require rinvex/laravel-subscriptions

Using version ^3.0 for rinvex/laravel-subscriptions
./composer.json has been updated
Loading composer repositories with package information
[...]
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi
Discovered Package: artesaos/seotools
Discovered Package: barryvdh/laravel-debugbar
Discovered Package: barryvdh/laravel-ide-helper
Discovered Package: facade/ignition
Discovered Package: felixkiss/uniquewith-validator
Discovered Package: fideloper/proxy
Discovered Package: intervention/image
Discovered Package: laravel/tinker
Discovered Package: laravel/ui
Discovered Package: laravelcollective/html
Discovered Package: laravelista/ekko
Discovered Package: nesbot/carbon
Discovered Package: nunomaduro/collision
Discovered Package: rinvex/laravel-subscriptions
Discovered Package: spatie/laravel-backup
Discovered Package: spatie/laravel-cookie-consent
Discovered Package: spatie/laravel-cors
Discovered Package: spatie/laravel-csp
Discovered Package: spatie/laravel-permission
Discovered Package: spatie/laravel-schemaless-attributes
Discovered Package: spatie/laravel-sitemap
Discovered Package: spatie/laravel-translatable
Package manifest generated successfully.
> Illuminate\Foundation\ComposerScripts::postUpdate

:$ php artisan rinvex:publish:subscriptions

***************************************************
*     Publish Rinvex Subscriptions Resources.     *
***************************************************

Unable to locate publishable resources.
Publishing complete.
Unable to locate publishable resources.
Publishing complete.
```

As you can see, package is discovered correctly, but service provider are not loaded:

```shell script
:$ php artisan vendor:publish

 Which provider or tag's files would you like to publish?:
  [0 ] Publish files from all providers and tags listed below
  [1 ] Provider: Artesaos\SEOTools\Providers\SEOToolsServiceProvider
  [2 ] Provider: Barryvdh\Debugbar\ServiceProvider
  [3 ] Provider: Barryvdh\LaravelIdeHelper\IdeHelperServiceProvider
  [4 ] Provider: Facade\Ignition\IgnitionServiceProvider
  [5 ] Provider: Fideloper\Proxy\TrustedProxyServiceProvider
  [6 ] Provider: Illuminate\Foundation\Providers\FoundationServiceProvider
  [7 ] Provider: Illuminate\Mail\MailServiceProvider
  [8 ] Provider: Illuminate\Notifications\NotificationServiceProvider
  [9 ] Provider: Illuminate\Pagination\PaginationServiceProvider
  [10] Provider: Intervention\Image\ImageServiceProviderLaravelRecent
  [11] Provider: Laravel\Tinker\TinkerServiceProvider
  [12] Provider: Laravelista\Ekko\Frameworks\Laravel\ServiceProvider
  [13] Provider: Spatie\Backup\BackupServiceProvider
  [14] Provider: Spatie\CookieConsent\CookieConsentServiceProvider
  [15] Provider: Spatie\Cors\CorsServiceProvider
  [16] Provider: Spatie\Csp\CspServiceProvider
  [17] Provider: Spatie\Permission\PermissionServiceProvider
  [18] Provider: Spatie\Sitemap\SitemapServiceProvider
  [19] Provider: Spatie\Translatable\TranslatableServiceProvider
  [20] Tag: config
  [21] Tag: flare-config
  [22] Tag: ignition-config
  [23] Tag: lang
  [24] Tag: laravel-errors
  [25] Tag: laravel-mail
  [26] Tag: laravel-notifications
  [27] Tag: laravel-pagination
  [28] Tag: migrations
  [29] Tag: views
 > ^C
```

I have tried a `composer dump-autoload` command, but nothing changes..
And I have tried even to manually add service provider to `config/app.php`, without success....

Any ideas??

**Comment by BoGnY** [comment]
@Omranic ?

**Comment by mscart** [comment]
same problem here ...

**Comment by mscart** [comment]
@BoGnY i find the bug
first of all your env must be local 
APP_ENV=local
then in rinvex/laravel-support/src/Traits/ConsoleTools.php functions 
protected function publishesResources(): bool
    {
        return  ! $this->app->environment('production');
    }

must be changed in :

protected function publishesResources(): bool
    {
        return  $this->app->environment('production');
    }

and 
protected function autoloadMigrations(string $config): bool
    {
        return  ! $this->app->environment('production') && $this->app['config'][$config.'.autoload_migrations'];
    }

in 
protected function autoloadMigrations(string $config): bool
    {
        return  $this->app->environment('production') && $this->app['config'][$config.'.autoload_migrations'];
    }
have a nice coding ! :)

**Comment by BoGnY** [comment]
this is not correct, resources **must not** be published in `production` environment!! with your changes, resources is published **only** on `production` environment...

the problem is that resources path are not resolved correctly, but the publishing function is correct!

**Comment by Omranic** [comment]
Thank you for your report. This has been fixed in the latest v4.0.4 release.
